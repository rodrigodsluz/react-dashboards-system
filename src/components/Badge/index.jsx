import React from 'react';

import PropTypes from 'prop-types';

import Theme from '../../theme';

const Badge = ({
  content, textColor, backgroundColor,
}) => (
  <>
    <span style={{
      display: 'inline-block',
      padding: '.4em',
      minWidth: '128px',
      fontSize: '90%',
      borderRadius: '30px',
      fontWeight: 400,
      color: textColor,
      backgroundColor,
      textAlign: 'center',
    }}
    >
      {content}
    </span>
  </>
);

Badge.defaultProps = {
  // textSize: 12,
  textColor: Theme.palette.text.light,
  backgroundColor: Theme.palette.primary.main,
};

Badge.propTypes = {
  content: PropTypes.string.isRequired,
  // textSize: PropTypes.number,
  textColor: PropTypes.string,
  backgroundColor: PropTypes.string,
};

export default Badge;
