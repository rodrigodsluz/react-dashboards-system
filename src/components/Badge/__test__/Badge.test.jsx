import React from 'react';
import renderer from 'react-test-renderer';

import Badge from '../index';

it('renders badge', () => {
  const tree = renderer
    .create(
      <Badge content="test" textColor="#FFFFFF" backgroundColor="#ff0000" />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
