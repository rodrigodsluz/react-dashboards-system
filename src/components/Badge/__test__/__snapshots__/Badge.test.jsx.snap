// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders badge 1`] = `
<span
  style={
    Object {
      "backgroundColor": "#ff0000",
      "borderRadius": "30px",
      "color": "#FFFFFF",
      "display": "inline-block",
      "fontSize": undefined,
      "fontWeight": 600,
      "padding": "3px 15px 3px 15px",
    }
  }
>
  test
</span>
`;
